src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -387,30 +387,37 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
 
         private void AddTupleTypeName(INamedTypeSymbol symbol)
         {
-            // TODO: revise to generate user-friendly name 
-            var members = string.Join(", ", symbol.GetMembers().OfType<IFieldSymbol>().Select(CreateTupleTypeMember));
+            var tupleType = (TupleTypeSymbol)symbol;
+            bool hasNames = tupleType.HasFriendlyNames;
 
-            if (members.Length == 0)
-            {
-                builder.Add(new SymbolDisplayPart(SymbolDisplayPartKind.ClassName, symbol, "<empty tuple type>"));
-            }
-            else
+            AddPunctuation(SyntaxKind.OpenParenToken);
+
+            bool first = true;
+            foreach (var member in tupleType.GetMembers().OfType<IFieldSymbol>())
```

Comment:
I think we should simply add a property to TupleTypeSymbol that would return the fields array.